source/app-development/tutorial-ps-to-quota.rst

```diff
@@ -0,0 +1,451 @@
+    -    assert_equal "28128", p.rss
+    -    assert_equal "0:00", p.time
+    -    assert_equal "Passenger RackApp: /users/PZS0562/efranz/ondemand/dev/quota", p.command
+    +    assert_equal "10.11.200.32:/PZS0562/", q.filesystem, "expected filesystem value not correct"
+    +    assert_equal "99616M", q.blocks, "expected blocks value not correct"
+    +    assert_equal "500G", q.blocks_limit, "expected blocks_limit value not correct"
+    +    assert_equal "933k", q.files, "expected files value not correct"
+    +    assert_equal "0", q.files_grace, "expected files_grace value not correct"
+      end
+
+
+Resulting test method:
+
+.. code:: ruby
+
+    class TestCommand < Minitest::Test
+
+      def test_command_output_parsing
+        output = <<-EOF
+    Disk quotas for user efranz (uid 10851):
+        Filesystem  blocks   quota   limit   grace   files   quota   limit   grace
+    10.11.200.32:/PZS0562/  99616M    500G    500G       0    933k   1000k   1000k       0
+    EOF
+        quotas = Command.new.parse(output)
+
+        assert_equal 1, quotas.count, "number of structs parsed should equal 1"
+
+        q = quotas.first
+
+        assert_equal "10.11.200.32:/PZS0562/", q.filesystem, "expected filesystem value not correct"
+        assert_equal "99616M", q.blocks, "expected blocks value not correct"
+        assert_equal "500G", q.blocks_limit, "expected blocks_limit value not correct"
+        assert_equal "933k", q.files, "expected files value not correct"
+        assert_equal "0", q.files_grace, "expected files_grace value not correct"
+      end
+    end
+
+Update command.rb
+.................
+
+Run test by running `rake` command and you will see it fail:
+
+.. code:: sh
+
+    $ rake
+    Run options: --seed 58990
+
+    # Running:
+
+    F
+
+    Finished in 0.000943s, 1060.4569 runs/s, 1060.4569 assertions/s.
+
+      1) Failure:
+    TestCommand#test_command_output_parsing [/users/PZS0562/efranz/ondemand/dev/quota/test/test_command.rb:14]:
+    number of structs parsed should equal 1.
+    Expected: 1
+      Actual: 3
+
+    1 runs, 1 assertions, 1 failures, 0 errors, 0 skips
+    rake aborted!
+    Command failed with status (1)
+
+    Tasks: TOP => default => test
+    (See full trace by running task with --trace)
+
+.. warning::
+
+   To run commands like rake through the shell you need to make sure you are on
+   a host that has the correct version of Ruby installed. For OnDemand that likely
+   means using Software Collections with the same packages used to install OnDemand.
+
+   With SCL, running rake with rh-ruby22 pacakge looks like:
+
+   ``scl enable rh-ruby22 -- rake``
+
+   With SCL, running git commands using git19 looks like:
+
+   ``scl enable git19 -- git commit -m "initial commit"``
+
+   You can avoid this by loading the SCL packages in your .bashrc or .bash_profile file.
+   For example, in my .bash_profile I have:
+
+   .. code:: sh
+
+      if [[ ${HOSTNAME%%.*} == webtest04*  ]]
```

Comment:
Maybe mention above that `webtest04` should be replaced with the hostname of the system they are using.